api/modules/leaderboards/leaderboards.service.js

```diff
@@ -0,0 +1,100 @@
+const { responseHelper } = require("../../helper/responses.helper");
+
+module.exports.leaderBoardsService = {
+  getLeaderBoards: (db, limit) => {
+    console.log("leader board reach");
+    return new Promise((resolve, reject) => {
+      let sql =
+        "select u.user_name, u.user_id, coalesce((select uh.highscore from user_highscore uh where uh.user_id = u.user_id), 0) as highscore, (select count(1)+1 from user_highscore where highscore > (select coalesce(highscore,0) from user_highscore uh where uh.user_id = u.user_id)) as rank from users u limit ?";
+
+      db.all(sql, [limit], (error, data) => {
+        if (error) {
+          reject(error);
+          console.log(error);
+        } else {
+          console.log(data);
+          resolve(data);
+        }
+      });
+    })
+      .then((leaderboardData) => {
+        return responseHelper.responseMapper(leaderboardData);
+      })
+      .catch((error) => {
+        return { error };
+      });
+  },
+  getUserRank: (db, userId) => {
+    return new Promise((resolve, reject) => {
+      let sql =
+        "select count(user_highscore_id)+1 as rank from user_highscore where highscore > (select coalesce(highscore,0) from user_highscore where user_id = ?)";
+      db.all(sql, [userId], (err, rankData) => {
+        if (err) {
+          reject(err);
+        } else {
+          resolve(rankData);
+        }
+      });
+    })
+      .then((userRankData) => {
+        return userRankData[0].rank;
+      })
+      .catch((err) => {
+        console.log(err);
+        return err;
+      });
+  },
+  getUserScore: (db, userId) => {
+    return new Promise((resolve, reject) => {
+      let sql =
+        "select user_id, highscore, user_highscore_id from user_highscore where user_id = ?";
+      db.all(sql, [userId], (err, data) => {
```

Comment:
There's a lot of this duplicated db.run and db.all stuff which takes the function that handles resolve/reject cases... why not just declare  `function(err, data) { return  new Promise(...` to "promisify" DB.all  and all.